src/main/java/org/mamute/dao/QuestionDAO.java

```diff
@@ -278,5 +277,126 @@ public Question questionWith(Attachment attachment) {
 				.setParameter("a", attachment)
 				.uniqueResult();
 	}
-}
 
+	public void delete(Question question) {
```

Comment:
It sounds fair enough. But there's another issue that rises if we follow this idea. We have some raw SQL queries in our code, right? We would have to check all of them and make sure to add a `where deleted=false` everywhere. Is that ok for you?